### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by RMeli**

### CODE REVIEW

**Comment by RMeli** [comment]
`mlflow<2.16` seems to work: https://github.com/pytorch/ignite/pull/3278

**Comment by RMeli** [comment]
Locally the `lr` test passes:
```
>>> Train Results - Epoch[1] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.65863
    ROC AUC: 1.00000
    Loss: 0.65863
    Epoch Time: 3.56563
    Elapsed Time: 3.56565
    Learning rate: 0.1
>>> Test Results - Epoch[1] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.65863
    ROC AUC: 1.00000
    Loss: 0.65863
>>> Train Results - Epoch[2] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.62210
    ROC AUC: 1.00000
    Loss: 0.62210
    Epoch Time: 3.48158
    Elapsed Time: 7.04723
    Learning rate: 0.1
>>> Test Results - Epoch[2] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.62210
    ROC AUC: 1.00000
    Loss: 0.62210
>>> Train Results - Epoch[3] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.52257
    ROC AUC: 1.00000
    Loss: 0.52257
    Epoch Time: 3.76639
    Elapsed Time: 10.81362
    Learning rate: 0.010000000000000002
>>> Test Results - Epoch[3] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.52257
    ROC AUC: 1.00000
    Loss: 0.52257
>>> Train Results - Epoch[4] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.47190
    ROC AUC: 1.00000
    Loss: 0.47190
    Epoch Time: 3.39351
    Elapsed Time: 14.20714
    Learning rate: 0.010000000000000002
>>> Test Results - Epoch[4] <<<
    Accuracy: 0.66667
    Balanced Accuracy: 0.50000
    Pose Loss: 0.47190
    ROC AUC: 1.00000
    Loss: 0.47190
>>> Train Results - Epoch[5] <<<
    Accuracy: 1.00000
    Balanced Accuracy: 1.00000
    Pose Loss: 0.36441
    ROC AUC: 1.00000
    Loss: 0.36441
    Epoch Time: 2.64669
    Elapsed Time: 16.85383
    Learning rate: 0.0010000000000000002
>>> Test Results - Epoch[5] <<<
    Accuracy: 1.00000
    Balanced Accuracy: 1.00000
    Pose Loss: 0.36441
    ROC AUC: 1.00000
    Loss: 0.36441
```
The loss appears to be the same for the first three epochs, but the `lr` is does not change at the third epoch.

**Comment by RMeli** [comment]
This is deprecated and needs to be removed:
https://github.com/RMeli/gnina-torch/blob/83c4b667fe9b87e5dad7fcaa05bb4baa8fc0320c/gninatorch/training.py#L890

**Comment by RMeli** [comment]
Updated the GitHub actions, and made the `lr` test longer (10 epochs), so that there is more chance to get two changes of the learning rate. The test remains potentially unstable.

**Comment by RMeli** [comment]
Actually, the `lr` test issues might have been caused by https://github.com/RMeli/gnina-torch/pull/57, which I totally forgot about.
